### PR TITLE
Fix API docs autogeneration for lazy loaded subpackages

### LINKED ISSUE #6074: API docs for the dev version are currently missing the filter module
## Description

I noticed today that the `filters` module missing from the API section of the dev docs. It shows up correctly in the 0.18.x ones. Specifically it is not listed on this page:

https://scikit-image.org/docs/dev/api/api.html

It is not just that the link is missing at the top-level API page. https://scikit-image.org/docs/dev/api/skimage.filters.html also gives a 404.

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #6074

Our apigen.py utility script that autogenerates the API docs was incompatible with the use of `lazy.attach` in our filters and data submodules. This PR updates this utility script.

Can verify by checking that the API pages for `filters` and `data` are now present in the documentation artifact.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
should backport to v0.19.x as well

**Comment by grlee77** [comment]
attn: @stefanv as this relates to lazy loading and may need similar considerations for other projects where it is being considered

**Comment by grlee77** [comment]
Okay, the pages are now there in the artifacts:
https://3220-2014929-gh.circle-artifacts.com/0/doc/build/html/api/skimage.data.html
https://3220-2014929-gh.circle-artifacts.com/0/doc/build/html/api/skimage.filters.html
and links to them show up at:
https://3220-2014929-gh.circle-artifacts.com/0/doc/build/html/api/api.html

**Comment by grlee77** [comment]
should be good to go if anyone is available to review

**Comment by stefanv** [comment]
Why not just use getattr consistently, then you don't need to make a special exception for lazy loaded modules?

**Comment by grlee77** [comment]
Yeah, that was apparently too easy :). I updated it now

**Comment by grlee77** [comment]
the artifact still looks good after the update

**Comment by grlee77** [comment]
@meeseeksdev backport to v0.19.x

File: doc/tools/apigen.py [hunk 1]
```diff
@@ -211,9 +211,10 @@ def _parse_module_with_import(self, uri):
         submodules = []
         for obj_str in obj_strs:
             # find the actual object from its string representation
-            if obj_str not in mod.__dict__:
+            try:
+                obj = getattr(mod, obj_str)
+            except AttributeError:
```
Comment (stefanv) [inline doc/tools/apigen.py:216]: :zany_face: That definitely took me by surprise!

**Review by stefanv — APPROVED** [review]

**Pull request merged**